.github/workflows/test_publish.yml

```diff
@@ -0,0 +1,28 @@
+name: CI
+
+on:
+  push:
+    branches: [main]
```

Comment:
Won't this try to publish on every push to main? You can't re-upload a version of a package, so will this fail?

Reply:
Also a good catch, this should just be on pull-requests. That said, the `skip_existing` flag should prevent it from failing (in test pypi) even if the same version exists. the test is really just to make sure it *could* be uploaded (like the dist/ folder is ok) so if it skips the actual upload because the version is the same it's NBD